### PR TITLE
Get spec_interval string from CarbonInterval

### LINKED ISSUE #525: Ability to get interval_spec from CarbonInterval
This is a feature request that I am happy to implement and send a Pull request with, but before that I'd like to discuss it.

I am currently in a need of getting `interval_spec` from `CarbonInterval` object. I imagine it could be achieved by implementing `spec` method on the `CarbonInterval` class. Do you see any potential issues with this?

### PR DESCRIPTION
**Pull request by martinssipenko**
Fixes #525

### CODE REVIEW

**Comment by martinssipenko** [comment]
Looks like CI is failures are false positives.

**Comment by dereuromark** [comment]
What about __toString returning this string?

    $this->assertEquals('PT1M', (string)$ci);

For me spec() doesn't say much about what it does or returns.

File: composer.json [hunk 1]
```diff
@@ -36,5 +36,10 @@
     "psr-4": {
       "Tests\\": "tests/"
     }
+  },
+  "extra": {
+      "branch-alias": {
+          "dev-master": "1.x-dev"
```
Comment (GrahamCampbell) [inline composer.json]: don't add this

Comment (GrahamCampbell) [inline composer.json]: move to a different PR

File: src/Carbon/CarbonInterval.php [hunk 2]
```diff
@@ -520,4 +520,39 @@ public function add(DateInterval $interval)
...
+        }
+
+        if (count($time) > 0) {
+            $specString .= static::PERIOD_TIME_PREFIX;
+            foreach ($time as $key => $value) {
+                $specString .= $value . $key;
+            }
+        }
+
+        return $specString === static::PERIOD_PREFIX ? 'PT0S' : $specString ;
```
Comment (dereuromark) [inline src/Carbon/CarbonInterval.php]: looks like there is a whitespace too much at the very end (before ;).

**Comment by martinssipenko** [comment]
Re: `__toString`, it would make it not backwards compatible.

**Comment by martinssipenko** [comment]
Closing this PR and moving code change to https://github.com/briannesbitt/Carbon/pull/530, that removes the updates in `composer.json`. Have to create a new PR, because my forks `master` branch is used by other apps i'm working on, and I have to keep the `composer.json` change in there.
